src/assisted_test_infra/test_infra/controllers/node_controllers/oci_controller.py

```diff
@@ -0,0 +1,149 @@
+from enum import Enum
+from typing import List, Tuple, Union
+
+import oci
+from oci.core import VirtualNetworkClient
+from oci.core.models import Instance
+
+import consts
+from assisted_test_infra.test_infra import BaseClusterConfig
+from assisted_test_infra.test_infra.controllers.node_controllers.tf_controller import TFController
+from assisted_test_infra.test_infra.helper_classes.config import BaseNodesConfig
+from assisted_test_infra.test_infra.helper_classes.config.base_oci_config import BaseOciConfig
+from service_client import log
+
+
+class OciInstanceState(Enum):
+    RUNNING = "RUNNING"
+    STARTING = "STARTING"
+    STOPPING = "STOPPING"
+    STOPPED = "STOPPED"
+    TERMINATING = "TERMINATING"
+    TERMINATED = "TERMINATED"
+    RESETTING = "RESETTING"
+    RESTARTING = "RESTARTING"
+
+
+class OciInstanceAction(Enum):
+    START = "START"
+    STOP = "STOP"
+    SOFTRESET = "SOFTRESET"
+
+
+class OciController(TFController):
+    _config: BaseOciConfig
+
+    def __init__(self, config: BaseNodesConfig, cluster_config: BaseClusterConfig):
+        super().__init__(config, cluster_config)
+        self._virtual_network_client: VirtualNetworkClient = None
+
+    def get_all_vars(self):
+        tfvars = super(OciController, self).get_all_vars()
+        tfvars["unique_id"] = self._entity_config.entity_name.suffix
+        tfvars["master_memory_gib"] = self._config.master_memory / consts.MiB_UNITS
+        tfvars["worker_memory_gib"] = self._config.worker_memory / consts.MiB_UNITS
+        tfvars["master_disk_size_gib"] = self._config.master_disk / consts.GB
+        if self._config.worker_disk == consts.resources.DEFAULT_WORKER_DISK:
+            self._config.worker_disk = 50 * consts.GB
```

Comment:
```suggestion
        # Minimal disk size on OCI is 50GB
        if self._config.worker_disk == consts.resources.DEFAULT_WORKER_DISK:
            self._config.worker_disk = 50 * consts.GB
```
comment to clarify why